stdlib/src/builtin/string.mojo

```diff
@@ -1971,6 +1972,115 @@ struct String(
             )
         return String(buf^)
 
+    @always_inline
+    fn _format[
+        *Ts: Stringable
+    ](
+        inout self,
```

Comment:
Why is it necessary to mutate self in this method? In the implementation you create `res`, you could just return it.

Reply:
Yes, this review can be implemented (`borrowed` self and `return` a result)